File: src/dotnet/HQ.Abstractions/Points/GetPointsV1.cs
using HQ.Abstractions.Points;

public class GetPointsV1
{
    public class Point
    {
        public Guid? Id { get; set; }
        public Guid? ChargeCodeId { get; set; }
        public String? ChargeCode { get; set; }
        public String? ProjectName { get; set; }
        public Guid? ProjectId { get; set; }
        public int Sequence { get; set; }
        public bool Completed { get; set; }
    }
    public class Request
    {
        public Guid? Id { get; set; }
        public Guid StaffId { get; set; }
        public DateOnly Date { get; set; }
    }
    public class Response
    {
        public DateOnly Date { get; set; }
        public Guid StaffId { get; set; }
        public List<Point> Points { get; set; } = new List<Point>();
        public DateOnly DisplayDate { get; set; }
        public DateOnly PreviousDate { get; set; }
        public DateOnly NextDate { get; set; }
    }
}

File: src/angular/hq/src/app/models/Points/get-points-v1.ts
export interface getPointsRequestV1 {
  id?: string | null;
  date: string;
  staffId: string;
}

export interface PlanningPoint {
  id: string | null;
  chargeCodeId: string | null;
  chargeCode: string | null;
  projectName: string | null;
  projectId: string | null;
  sequence: number;
}

export interface getPointsResponseV1 {
  date: string;
  staffId: string;
  points: PlanningPoint[];
  displayDate: string;
  previousDate: string;
  nextDate: string;
}
